File: 6kyu/Are they the same/comp.js
function comp(array1, array2) {
  if (array1 && array2) {
    if (array1.length === 0 && array2.length === 0) {
      return true
    } else if (array1.length === 0 || array2.length === 0) {
      return false
    } else {
      array1 = array1.sort((a, b) => a - b)
      array2 = array2.sort((a, b) => a - b)
      for (let i = 0; i < array1.length; i++) {
        if (array2[i] !== array1[i] * array1[i]) {
          return false
        }
      }

      return true
    }
  }
  return false
}

File: 6kyu/Are they the same/comp.py
import math


def comp(array1, array2):
    if len(array1) == 0 and len(array2) == 0:
        return True
    elif len(array1) == 0 or len(array2) == 0:
        return False
    else:
        array1.sort()
        array2.sort()
        print(array1, array2)
        for i in range(len(array1)):
            print(i)
            if(array1[i]*array1[i] != array2[i]):
                return False
            pass
        return True
    # your code


a1 = [121, 144, 19, 161, 19, 144, 19, 11]
a2 = [121, 14641, 20736, 361, 25921, 361, 20736, 361]
print comp(a1, a2)
